### DISCUSSION TITLE
Cannot install homebrew on Catalina 10.15.7

### DISCUSSION THREAD

**Discussion by joeynorkin** [Homebrew/discussions#201] [Getting started]
This isn't my first Mac and just like all previous Macs of mine, one of the first things I do is install homebrew (after I install xcode command-line-tools of course). For this first time, this didn't work out. 

I ran `/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"` and the output I got was:

> $ /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
> Password:
> ==> This script will install:
> /usr/local/bin/brew
> /usr/local/share/doc/homebrew
> /usr/local/share/man/man1/brew.1
> /usr/local/share/zsh/site-functions/_brew
> /usr/local/etc/bash_completion.d/brew
> /usr/local/Homebrew
> ==> The following existing directories will be made group writable:
> /usr/local/lib
> ==> The following existing directories will have their owner set to <_edited out_>:
> /usr/local/lib
> ==> The following existing directories will have their group set to admin:
> /usr/local/lib
> ==> The following new directories will be created:
> /usr/local/etc
> /usr/local/include
> /usr/local/var
> /usr/local/opt
> /usr/local/var/homebrew
> /usr/local/var/homebrew/linked
> /usr/local/Cellar
> /usr/local/Caskroom
> /usr/local/Homebrew
> /usr/local/Frameworks
> 
> Press RETURN to continue or any other key to abort
> ==> /usr/bin/sudo /bin/chmod u+rwx /usr/local/lib
> ==> /usr/bin/sudo /bin/chmod g+rwx /usr/local/lib
> chmod: Unable to change file mode on /usr/local/lib: Operation not permitted
> Failed during: /usr/bin/sudo /bin/chmod g+rwx /usr/local/lib

I tried adding Terminal to Full Disk Access and also tried disabling my SIP. Neither of which has worked. Any ideas? This is a work computer just in case you needed to know.

Thanks.

**Comment by carlocab** [comment]
Have you tried setting the permissions of the contents of `/usr/local` manually? You can do this with

    sudo chown -R <your_username>:admin /usr/local/*

**Reply by joeynorkin** [reply]
I've tried that before but just in case I got something wrong, I tried it again with the following output:

 

> $ sudo chown -R \`whoami\`:admin /usr/local/*
> Password:
> chown: /usr/local/lib: Operation not permitted
> chown: /usr/local/sbin/wepsvc: Operation not permitted

**Reply by joeynorkin** [reply]
In case this helps, here's the output for `ls -la /usr/local`
```
$ ls -la /usr/local/
drwxr-xr-x   9 root         wheel  288 Nov 25 10:55 .
drwxr-xr-x@ 11 root         wheel  352 Nov 25 10:52 ..
-rw-r--r--   1 <user_name>  wheel    0 Nov 25 09:41 .com.apple.installer.keep
drwxrwxr-x   5 <user_name>  admin  160 Nov 24 17:52 bin
drwxr-xr-x   4 <user_name>  admin  128 Nov 24 17:52 fp
drwxr-xr-x   3 <user_name>  admin   96 Nov 24 17:51 jamf
drwxr-xr-x   5 root         wheel  160 Nov 24 17:52 lib
drwxrwxr-x   4 <user_name>  admin  128 Nov 24 17:52 sbin
drwxrwxr-x   3 <user_name>  admin   96 Nov 24 17:51 share
```

**Reply by SMillerDev** [reply]
It could very well be that this is blocked by Jamf. `<user_name>:admin` are the default permissions for anything in `/usr/local` so something has altered these.

**Reply by joeynorkin** [reply]
I verified that this is not a reproducible issue and that it had to do with security software for work. Thanks everyone!

**Comment by carlocab** [comment]
What’s in `/usr/local/lib`?

**Reply by joeynorkin** [reply]
```
$ ls -la lib/
total 1280
drwxr-xr-x  5 root         wheel     160 Nov 24 17:52 .
drwxr-xr-x  9 root         wheel     288 Nov 25 10:55 ..
-rwxr-xr-x  1 <user_name>  admin   49456 Oct 11  2019 libMIME_Extractor.dylib
-rwxr-xr-x  1 <user_name>  admin  430064 Oct 11  2019 libwep
-rwxr-xr-x  1 <user_name>  admin  169408 Oct 11  2019 libwep_chrome.dylib
```

**Reply by gromgit** [reply]
What's the output of `/bin/ls -ld@eO /usr/local/lib /usr/local/sbin/wepsvc`?

**Reply by joeynorkin** [reply]
```
$ /bin/ls -ld@eO /usr/local/lib /usr/local/sbin/wepsvc
drwxr-xr-x  5 root  wheel  -    160 Nov 24 17:52 /usr/local/lib
-rwxr-xr-x  1 root  wheel  - 307696 Oct 11  2019 /usr/local/sbin/wepsvc
```

**Reply by ByronGary** [reply]
Did you get this sorted out? i'm having the same issue

**Reply by gromgit** [reply]
As @joeynorkin mentioned in a separate thread:
> I verified that this is not a reproducible issue and that it had to do with security software for work.

Unless you're running some security software that's quietly resetting directory permissions and/or ownerships, you almost certainly have a different issue.

Have you already tried what @carlocab mentioned?
```
sudo chown -R <your_username>:admin /usr/local/*
```

**Reply by ThomasMatt** [reply]
I'm getting the same error also..

I tried: sudo chown -R thomasmatt:admin /usr/local/*
But all I get is "Operation Not Permitted"

**Comment by ThomasMatt** [comment]
I seem to be having near the same issues. I have no security software installed on the mac and I am the Admin.

I've tried most of the steps above also ^^


This is what I get when trying to install:

`
Press RETURN to continue or any other key to abort
==> /usr/bin/sudo /bin/chmod u+rwx /usr/local/bin /usr/local/etc /usr/local/include /usr/local/lib /usr/local/sbin /usr/local/share /usr/local/opt /usr/local/var /usr/local/Frameworks /usr/local/etc/bash_completion.d /usr/local/lib/pkgconfig /usr/local/share/aclocal /usr/local/share/doc /usr/local/share/info /usr/local/share/locale /usr/local/share/man /usr/local/share/man/man1 /usr/local/share/man/man3 /usr/local/share/man/man5 /usr/local/share/man/man8 /usr/local/share/zsh /usr/local/share/zsh/site-functions
==> /usr/bin/sudo /bin/chmod g+rwx /usr/local/bin /usr/local/etc /usr/local/include /usr/local/lib /usr/local/sbin /usr/local/share /usr/local/opt /usr/local/var /usr/local/Frameworks /usr/local/etc/bash_completion.d /usr/local/lib/pkgconfig /usr/local/share/aclocal /usr/local/share/doc /usr/local/share/info /usr/local/share/locale /usr/local/share/man /usr/local/share/man/man1 /usr/local/share/man/man3 /usr/local/share/man/man5 /usr/local/share/man/man8
==> /usr/bin/sudo /bin/chmod 755 /usr/local/share/zsh /usr/local/share/zsh/site-functions
chmod: Unable to change file mode on /usr/local/share/zsh: Operation not permitted
chmod: Unable to change file mode on /usr/local/share/zsh/site-functions: Operation not permitted
Failed during: /usr/bin/sudo /bin/chmod 755 /usr/local/share/zsh /usr/local/share/zsh/site-functions`

**Comment by deepashree-dexcom** [comment]
`sudo: 4294967295: invalid value
sudo: error initializing audit plugin sudoers_audit
Failed during: //usr/bin/sudo /usr/sbin/chown -R <username>:admin /usr/local/Homebrew`

This is the error that I am getting when I try to install homebrew on my macbook running Catalina OS. 
Any tips ?

**Reply by gromgit** [reply]
Did you literally run:
```
/usr/bin/sudo /usr/sbin/chown -R <username>:admin /usr/local/Homebrew
```
or did you substitute your actual user name, e.g.:
```
/usr/bin/sudo /usr/sbin/chown -R deepashree:admin /usr/local/Homebrew
```

**Reply by deepashree-dexcom** [reply]
I received this error when I attempted to install Homebrew and it did have my actual username.

**Reply by gromgit** [reply]
A quick web search found the following: https://discussions.apple.com/thread/252518458

What's the output of `id` on your system? Are you binding to an Active Directory server on your machine?

**Reply by deepashree-dexcom** [reply]
The issue is resolved this morning. 
Last night I edited sudoers file using 'sudo visudo' . I added following line to it.
`<username>          ALL = (ALL) ALL`
It did not do any magic last night even after reboot. 
This morning I rebooted the machine again and installation started . 
I only had to do 
`sudo chown -R $(whoami) /usr/local/Cellar`

**Comment by ahmedhossam91** [comment]
is there any way to install brew offline ?

**Reply by gromgit** [reply]
Asked and answered at https://github.com/Homebrew/discussions/discussions/1417#discussioncomment-711782
